src/supplicant/nm-supplicant-config.c

```diff
@@ -452,13 +452,13 @@ gboolean
 nm_supplicant_config_add_setting_wireless (NMSupplicantConfig * self,
                                            NMSettingWireless * setting,
                                            guint32 fixed_freq,
+                                           const GByteArray *ssid,
```

Comment:
`GByteArray` is a mutable array. This should be a `GBytes` type, which is immutable. I guess, you did this, because of `nm_wifi_ap_get_ssid()`. But it's really NMWifiAP who does it wrong already and should be fixed. See how even `nm_wifi_utils_complete_connection()` is wrong, and at first need to convert the `GByteArray` argument to a `GBytes`. Let's not proliferate the use of the ill suited type, but rather fix it.

Reply:
Another reason, `GBytes` being immutable, there is little reason to ever clone it, just take another ref. Avoiding pointless cloning at
```C
nm_wifi_ap_set_ssid (ap, g_bytes_get_data (ssid, NULL), g_bytes_get_size (ssid));
```

Reply:
I suppose this is what #182 does?